cmd/deploy/pipeline/control.go

```diff
@@ -0,0 +1,370 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/okteto/okteto/pkg/cmd/pipeline"
+	"github.com/okteto/okteto/pkg/constants"
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	"github.com/okteto/okteto/pkg/log/io"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/okteto/okteto/pkg/types"
+	"github.com/spf13/afero"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+const (
+	dependencyEnvTemplate = "OKTETO_DEPENDENCY_%s_VARIABLE_%s"
+)
+
+var (
+	errUnableToReuseParams = errors.New("development environment not found: unable to use --reuse-params option")
+)
+
+// Deployer deploys a pipeline
+type Deployer struct {
+	opts              *Options
+	fs                afero.Fs
+	pipelineCtrl      types.PipelineInterface
+	streamCtrl        types.StreamInterface
+	k8sClientProvider k8sClientProvider
+	okCtxController   okCtxController
+	ioCtrl            *io.IOController
+}
+
+type k8sClientProvider interface {
+	Provide(clientApiConfig *api.Config) (kubernetes.Interface, *rest.Config, error)
+}
+
+type okCtxController interface {
+	GetNamespace() string
+	GetK8sConfig() *api.Config
+}
+
+// NewPipelineDeployer returns a new pipeline deployer
+func NewPipelineDeployer(o ...OptFunc) (*Deployer, error) {
+	opts := &Options{}
+	for _, of := range o {
+		of(opts)
+	}
+	if err := opts.setDefaults(); err != nil {
+		return nil, fmt.Errorf("could not set default values for options: %w", err)
+	}
+	return &Deployer{
+		opts:              opts,
+		fs:                opts.fs,
+		k8sClientProvider: opts.k8sClientProvider,
+		okCtxController:   opts.okCtxController,
+		ioCtrl:            opts.ioCtrl,
+		pipelineCtrl:      opts.okClient.Pipeline(),
+		streamCtrl:        opts.okClient.Stream(),
+	}, nil
+}
+
+// Deploy deploys a pipeline
+func (d *Deployer) Deploy(ctx context.Context) error {
```

Comment:
current deploy method for pipeline command call `setDefault` method. Check [here](https://github.com/okteto/okteto/blob/master/cmd/pipeline/deploy.go#L143-L145). Why it is not added to this temporary duplicated code?